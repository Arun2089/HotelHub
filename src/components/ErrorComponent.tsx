import { Paper, Box, Typography } from "@mui/material";
import LoadingButton from "./LoadingButton";

type ErrorComponentProps = {
  className?: string;
  boxClassName?: string;
  onSubmit: () => void;
  message?: string;
  heading?: string;
  headingClassName?: string;
};

const ErrorComponent = ({
  className = "",
  onSubmit,
  message = "",
  boxClassName = "",
  heading,
  headingClassName = "",
}: ErrorComponentProps) => {
  return (
    <Paper
      className={`rounded-xl flex flex-col justify-center items-center ${className}`}
    >
      {heading && (
        <Typography
          className={`text-2xl self-start font-medium ${headingClassName}`}
        >
          {heading}
        </Typography>
      )}
      <Box
        className={`flex flex-grow flex-col justify-center items-center gap-y-2 ${boxClassName}`}
      >
        <Typography>{message}</Typography>
        <LoadingButton
          className="w-24"
          buttonText="Reload"
          onSubmit={onSubmit}
          size="small"
          color="error"
          variant="outlined"
        />
      </Box>
    </Paper>
  );
};

export default ErrorComponent;
