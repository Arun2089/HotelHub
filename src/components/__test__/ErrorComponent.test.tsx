import { render, screen, fireEvent, waitFor } from "@testing-library/react";
import "@testing-library/jest-dom";
import ErrorComponent from "../ErrorComponent";

const onSubmitMock = jest.fn();
const errorMessage = "Something went wrong";
const heading = "Error";

describe("ErrorComponent", () => {
  it("snapshot test", () => {
    const { asFragment } = render(
      <ErrorComponent
        onSubmit={onSubmitMock}
        message={errorMessage}
        heading={heading}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the error message and reload button", () => {
    render(
      <ErrorComponent
        onSubmit={onSubmitMock}
        message={errorMessage}
        heading={heading}
      />
    );

    const errorHeading = screen.getByText(heading);
    const errorMessageText = screen.getByText(errorMessage);
    const reloadButton = screen.getByRole("button", { name: "Reload" });

    expect(errorHeading).toBeInTheDocument();
    expect(errorMessageText).toBeInTheDocument();
    expect(reloadButton).toBeInTheDocument();
  });

  it("calls the onSubmit function when the reload button is clicked", async () => {
    render(
      <ErrorComponent
        onSubmit={onSubmitMock}
        message={errorMessage}
        heading={heading}
      />
    );

    const reloadButton = screen.getByRole("button", { name: "Reload" });
    fireEvent.click(reloadButton);
    await waitFor(() => expect(onSubmitMock).toHaveBeenCalledTimes(1));
  });
});
