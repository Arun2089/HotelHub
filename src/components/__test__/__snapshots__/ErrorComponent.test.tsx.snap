// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorComponent snapshot test 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 rounded-xl flex flex-col justify-center items-center  css-1ps6pg7-MuiPaper-root"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 text-2xl self-start font-medium  css-ahj2mt-MuiTypography-root"
    >
      Error
    </p>
    <div
      class="flex flex-grow flex-col justify-center items-center gap-y-2  MuiBox-root css-0"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        Something went wrong
      </p>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedError MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-root MuiButton-outlined MuiButton-outlinedError MuiButton-sizeSmall MuiButton-outlinedSizeSmall w-24 css-1ir7th6-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Reload
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
